### PR TITLE
Add tip about fixing commit mistakes in PR branches + minor grammar fix

### PR DESCRIPTION
**Pull request by puchik**
- Shortened and made minor grammar fixes to the third paragraph under "Updating your branch" in "PR Workflow"
- Added a tip regarding fixing commit mistakes in branches that are in pull requests.

These commands have been useful for me in the past but took some time to figure out how to use correctly. Feel free, however, to close if this is not the best approach to this issue. In any case, it would be useful for the documentation to mention how to proceed after making mistakes in forked repo branches.

### CODE REVIEW

**Comment by Calinou** [comment]
It should be noted that `git reset --hard` can be a dangerous operation, especially if some changes are untracked or uncommitted. However, if you've committed changes that you reset using `git reset --hard`, you may still be able to recover them by resetting to a commit ID found with the `git reflog` command.

**Comment by puchik** [comment]
I was thinking about that too, and that is a good explanation. I can add that directly under inside a note box. i.e.
"Note:
git reset --hard can be a dangerous operation, especially if you have untracked or uncommitted changes. However, if you've committed changes that you reset using git reset --hard, you may still be able to recover them by resetting to a commit ID found with the git reflog command."

File: community/contributing/pr_workflow.rst [hunk 1]
```diff
@@ -204,6 +204,20 @@ Therefore, the command that you should (almost) always use is:
 
     $ git pull --rebase upstream master
 
+
+If you have already pushed the merge commit without using ``rebase`` (or
```
Comment (NathanLovato) [inline community/contributing/pr_workflow.rst]: No need to parentheses here, just a comma:
`If you have already pushed the merge commit without using ``rebase``, or have made any mistakes that have resulted in undesired history, ...`

File: community/contributing/pr_workflow.rst [hunk 2]
```diff
@@ -204,6 +204,20 @@ Therefore, the command that you should (almost) always use is:
 
     $ git pull --rebase upstream master
 
+
+If you have already pushed the merge commit without using ``rebase`` (or 
+have made any mistakes that have resulted in undesired history), you may
+use ``reset`` to revert to a specific commit and try again:
```
Comment (NathanLovato) [inline community/contributing/pr_workflow.rst]: Suggestion: you may use a hard reset. Reset itself doesn't change the history. Like @Calinou suggested, it'd be nice to explain what a hard reset is and what it does as it's a destructive operation.

**Comment by NathanLovato** [comment]
Thanks for your contributions! I left you two comments to hopefully help improve it a bit.

**Comment by puchik** [comment]
Thanks! I've pushed the changes.

**Comment by akien-mga** [comment]
That's not the right place for such information IMO, there's a section further down on "Modifying a pull request" which explains how to interactively rebase a branch or amend a commit.

**Comment by puchik** [comment]
This tidbit is supposed to address merging updates from master, whereas "Modifying a pull request" applies more for making small changes to your own work. The difference is that in the former you may be dealing with hundreds of commits, and in the latter it may be one or two which are more manageable to manually pick/fixup.

Let me know what you think.

**Comment by akien-mga** [comment]
I see. Some explanations on how to reset one's branch to upstream/master can indeed be useful. I don't want it to be too verbose though, I'll have a closer look in coming days and see if I like the proposed text or if I would want any amend.

**Comment by mhilbrunner** [comment]
Having looked at the page in question, IMO this is a good addition and the right place for this information.

I'll be daring and merge this, if only to decrease Akiens workload a little, and seeing the timeline on this PR. If we eventually come up with *even further* improvements, we'll chan do those in a new PR.

Thank you, @puchik!

**Pull request merged**